packages/vscode-extension/src/treeviews/databaseInformationTreeDataProvider.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Event,
+  EventEmitter,
+  TreeDataProvider,
+  TreeItem,
+  TreeItemCollapsibleState,
+} from 'vscode';
+import {
+  getActiveConnection,
+  getDbSchemaInformation,
+} from '../connectionService';
+
+type DatabaseInformationItemType = 'label' | 'relationship';
+
+export class DatabaseInformationTreeDataProvider
+  implements TreeDataProvider<DatabaseInformationItem>
+{
+  private _onDidChangeTreeData: EventEmitter<
+    DatabaseInformationItem | undefined | void
+  > = new EventEmitter<DatabaseInformationItem | undefined | void>();
+  readonly onDidChangeTreeData: Event<
+    DatabaseInformationItem | undefined | void
+  > = this._onDidChangeTreeData.event;
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(element: DatabaseInformationItem): TreeItem | Thenable<TreeItem> {
+    return element;
+  }
+
+  getChildren(element?: DatabaseInformationItem): DatabaseInformationItem[] {
+    if (!element) {
+      return this.getActiveConnectionInformation();
+    }
+
+    return this.getDatabaseInformationElements(element);
+  }
+
+  private getActiveConnectionInformation(): DatabaseInformationItem[] | null {
+    const activeConnection = getActiveConnection();
+    if (!activeConnection) {
+      return null;
+    }
+
+    return [
+      new DatabaseInformationItem(
+        'label',
+        'Labels',
+        TreeItemCollapsibleState.Collapsed,
+      ),
+      new DatabaseInformationItem(
+        'relationship',
+        'Relationships',
+        TreeItemCollapsibleState.Collapsed,
+      ),
+    ];
+  }
+
+  private getDatabaseInformationElements(
+    element: DatabaseInformationItem,
+  ): DatabaseInformationItem[] {
+    const dbSchemaInformation = getDbSchemaInformation();
+
+    return element.type === 'label'
+      ? this.mapDbSchemaProperties('label', dbSchemaInformation.labels)
```

Comment:
Couldn't `dbSchemaInformation` be null here?

Reply:
yep, you're right - I'll add an undefined check here